core/src/main/java/tc/oc/pgm/loot/LootableModule.java

```diff
@@ -0,0 +1,125 @@
+package tc.oc.pgm.loot;
+
+import com.google.common.collect.ImmutableList;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.logging.Logger;
+import org.bukkit.inventory.ItemStack;
+import org.jdom2.Document;
+import org.jdom2.Element;
+import org.jetbrains.annotations.Nullable;
+import tc.oc.pgm.api.filter.Filter;
+import tc.oc.pgm.api.map.MapModule;
+import tc.oc.pgm.api.map.factory.MapFactory;
+import tc.oc.pgm.api.map.factory.MapModuleFactory;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.match.MatchModule;
+import tc.oc.pgm.api.module.exception.ModuleLoadException;
+import tc.oc.pgm.api.region.Region;
+import tc.oc.pgm.filters.FilterMatchModule;
+import tc.oc.pgm.filters.matcher.StaticFilter;
+import tc.oc.pgm.filters.parse.DynamicFilterValidation;
+import tc.oc.pgm.itemmeta.ItemModifyMatchModule;
+import tc.oc.pgm.util.TimeUtils;
+import tc.oc.pgm.util.compose.Composition;
+import tc.oc.pgm.util.compose.CompositionParser;
+import tc.oc.pgm.util.xml.InvalidXMLException;
+import tc.oc.pgm.util.xml.XMLUtils;
+
+public class LootableModule implements MapModule<LootableMatchModule> {
+
+  private static final Function<MapFactory, CompositionParser<ItemStack>> LOOT_PARSER_FACTORY =
+      factory ->
+          new CompositionParser<ItemStack>(factory) {
+            @Override
+            public ItemStack parseUnit(Element element) throws InvalidXMLException {
+              return this.factory.getKits().parseItemStack(element);
+            }
+          };
```

Comment:
I feel like this factory wouldn't be needed if the class wasnt annonymous.

Consider simply having an `LootParser extends CompositionParser<ItemStack> {`, and your "factory" can be simply calling: `new LootParser(mapFactory);`, making this Function<...> redundant